src/controllers/TorusController.ts

```diff
@@ -631,6 +632,44 @@ export default class TorusController extends BaseController<TorusControllerConfi
     }
   }
 
+  // burn NFT
+  async burnToken(NFTtoBurn: string, senderAddress: string) {
+    try {
+      const tx = new Transaction();
+      // get the publickey of the NFT
+      const mintPublickey = new PublicKey(NFTtoBurn);
+
+      // signer pub key
+      const signer = new PublicKey(senderAddress);
+
+      // determine the associated token account of the NFT
+      const associatedAddress = await getAssociatedTokenAddress(mintPublickey, signer);
```

Comment:
Do not depend on PDA for burning.
get the exact token account address for burning.
It is possible nft is not PDA

Reply:
find all token accounts by owner, then filter for the NFT mint address. Its possible that the associatedAddress isn't holding the mint account. 

optionally, you can check the accountInfo for :646 to verify the mint address. If that fails, you can resort to the above.